parquet_go18.go

```diff
@@ -27,12 +27,19 @@ func Read[T any](r io.ReaderAt, size int64, options ...ReaderOption) (rows []T,
 	}
 	rows = make([]T, file.NumRows())
 	reader := NewGenericReader[T](file, config)
-	n, err := reader.Read(rows)
-	if err == io.EOF {
-		err = nil
+	var i int
+	for {
+		c, err := reader.Read(rows[i:])
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return rows, err
+		}
+		i = c
```

Comment:
Hi, also been looking for a fix for this. I appreciate that someone else has been working on getting this fixed!

I'm a little concerned about this code - I believe that `reader.Read` returns the number of rows read. Shouldn't we be setting `i += c`?

Reply:
hi @grantwwu yes so sorry, good catch. updated just now